deploy/docker/Dockerfile

```diff
@@ -43,7 +43,7 @@ COPY deploy/docker/docker-entrypoint.sh deploy/docker/start.sh tmp/qemu-$QEMU_AR
 COPY --from=builder /emqx-rel/_build/$EMQX_NAME/rel/emqx /opt/emqx
 
 RUN ln -s /opt/emqx/bin/* /usr/local/bin/ 
-RUN apk add --no-cache curl ncurses-libs openssl sudo libstdc++
+RUN apk add --no-cache curl ncurses-libs openssl sudo libstdc++ bash perl
```

Comment:
#### Replace `sh` with `bash` for safety (prevent shell injection attack):
1. Use `compgen -e` to get all valid variable names.
  1.1 `env` will output the variable such as `$(attack)=value`, but compgen will ignore the invalid variable `$(attack)`.
  1.2 `for var in $(env)` will split the multi line value such as `export multiline=$(echo -e 'a\nmultiline\nvalue\n')`, it will be assumed as [`multiline=a`, `multiline`, `value`].
2. Use `${!var_name}` to get value from variable name safely.
  2.1 prevent injection attack.
  2.2 get multiline value.

#### Replace `sed` with `perl` in some scenes for read variable as raw string (prevent assume the input as regex):
1. use `-s` to pass variable into perl as raw string.
2. use quoted string `\Q$var\E` to escape input `$var` in regex.